// Generated by CoffeeScript 1.6.3
var argumentException, testable;

argumentException = require('../notice/errors').argumentException;

testable = void 0;

module.exports._middleware = function() {
  return testable;
};

module.exports.middleware = function(config) {
  var api, local;
  if (config == null) {
    config = {};
  }
  testable = local = {
    slots: {},
    nextSlot: function() {
      console.log({
        next: 1
      });
      return 1;
    },
    update: function(_arg, callback) {
      var description, enabled, fn, slot, title;
      slot = _arg.slot, title = _arg.title, description = _arg.description, enabled = _arg.enabled, fn = _arg.fn;
      if (slot == null) {
        slot = local.nextSlot();
      }
      if (typeof slot !== 'number') {
        throw argumentException('opts.slot', 'notice.use(opts, fn)', 'as number');
      }
      return console.log({
        UPDATE: slot
      });
    }
  };
  return api = {
    update: local.update
  };
};
